src/components/character-avatar/character-avatar.tsx

```diff
@@ -0,0 +1,48 @@
+import { Ash, Dot, Eve, Hal, Roy } from "~/assets/characters";
+import { type Character, type MatchStage } from "~/types";
+import { type FC } from "react";
+import { Stack } from "@mui/material";
+import { styles } from "./styles.js";
+
+interface Props {
+  character: Character;
+  stage?: MatchStage;
+  isSelected?: boolean;
+  hasGuessed?: boolean;
+  isBot?: boolean;
+  onSelectPlayer?: () => void;
+}
+
+export const CharacterAvatar: FC<Props> = ({
+  stage,
+  character,
+  isSelected,
+  isBot,
+  onSelectPlayer,
+}) => {
+  return (
+    <Stack
+      sx={styles.avatar(stage, isSelected, isBot)}
+      onClick={onSelectPlayer}
+    >
+      {getCharacter(character)}
+    </Stack>
+  );
+};
+
+const getCharacter = (character: Character) => {
```

Comment:
since you only need to tie the name to the component I think a map would be simpler here
```js
const character = {
  hal: <Hal />,
  ash: <Ash />,
  ...rest,
};
const Hal = character.hal;